v2/pkg/input/uncover.go

```diff
@@ -0,0 +1,152 @@
+package input
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/projectdiscovery/gologger"
+	"github.com/projectdiscovery/ratelimit"
+	ucRunner "github.com/projectdiscovery/uncover/runner"
+	"github.com/projectdiscovery/uncover/uncover"
+	"github.com/projectdiscovery/uncover/uncover/agent/censys"
+	"github.com/projectdiscovery/uncover/uncover/agent/fofa"
+	"github.com/projectdiscovery/uncover/uncover/agent/hunter"
+	"github.com/projectdiscovery/uncover/uncover/agent/quake"
+	"github.com/projectdiscovery/uncover/uncover/agent/shodan"
+	"github.com/projectdiscovery/uncover/uncover/agent/shodanidb"
+	"github.com/projectdiscovery/uncover/uncover/agent/zoomeye"
+)
+
+func GetTargetsFromUncover(delay, limit int, engine, query []string) (chan string, error) {
```

Comment:
move this logic as well to separate package

Reply:
moved this to uncover package